### PR TITLE
Convert all components in Calendar folder

### LINKED ISSUE #336: Update react-big-calendar
The version of `react-big-calendar` we currently use depends on a deprecated version of `popper.js`.

### PR DESCRIPTION
**Pull request by EricPedley**
## Summary
Child of #411. Converts everything in components/Calendar to TypeScript, as well as reorganizing the folder a little. Also, negative lines added balance on this one 😁
## Test Plan
Confirm it compiles and mess around with the calendar features a little bit
## Issues
Might close #336? @ecxyzzy  https://github.com/icssc/AntAlmanac/pull/420/commits/d33b688c22f14b994af8f7fc4cfed94b6d0b058f

### CODE REVIEW

**Comment by EricPedley** [comment]
The force push was because I accidentally merged main in too early. I want to wait until merging #411 to do the merge all at once. Also, I'm only gonna ask that one of you guys review before I merge; this change doesn't need everyone's input.

File: src/components/Calendar/Toolbar/CustomEventDialog/CustomEventDialog.tsx [hunk 1]
```diff
@@ -14,32 +14,58 @@ import {
...
+    color?: string
+}
 
-class CustomEventDialog extends PureComponent {
-    state = {
+interface CustomEventDialogProps {
+    customEvent?: CustomEvent,
+    onDialogClose?: ()=>void,
+    currentScheduleIndex: number,
+    scheduleNames: any[] // TODO: statically type this.
```
Comment (ChaseC99) [inline src/components/Calendar/Toolbar/CustomEventDialog/CustomEventDialog.tsx]:
```ts
scheduleNames: any[] // TODO: statically type this.
```

they aren't just strings?

File: src/components/Calendar/Toolbar/ExportCalendar.tsx [hunk 2]
```diff
@@ -228,13 +227,13 @@ const exportCalendar = () => {
...
-                startOutputType: 'local',
-                endOutputType: 'local',
+                startOutputType: 'local' as const,
+                endOutputType: 'local' as const,
                 title: `${deptCode} ${courseNumber} ${sectionType}`,
                 description: `${courseTitle}\nTaught by ${instructors.join('/')}`,
                 location: `${meeting.bldg}`,
-                start: firstClassStart,
-                end: firstClassEnd,
+                start: firstClassStart as [number,number,number,number,number],
```
Comment (ChaseC99) [inline src/components/Calendar/Toolbar/ExportCalendar.tsx]: I wonder if we should abstract this to it's own type alias, so it's more descriptive and we don't have to repeat `[number,number,number,number,number]` all over the place.

Something like: 

```ts
type DateTimeArray = [number,number,number,number,number];
```

File: src/components/Calendar/Toolbar/ExportCalendar.tsx [hunk 3]
```diff
@@ -75,29 +74,29 @@ const getClassStartDate = (term, bydays) => {
...
-    ];
+    ] as const;
 };
 
-// getFirstClass returns the start and end datetime of the first class
-//  Ex: ([2021, 3, 30], " 4:00-4:50p") -> [[2021, 3, 30, 16, 0], [2021, 3, 30, 16, 50]]
-const getFirstClass = (date, time) => {
+/** getFirstClass returns the start and end datetime of the first class
+    Ex: ([2021, 3, 30], " 4:00-4:50p") -> [[2021, 3, 30, 16, 0], [2021, 3, 30, 16, 50]] */
+const getFirstClass = (date: readonly [number, number, number], time: string) => {
```
Comment (ChaseC99) [inline src/components/Calendar/Toolbar/ExportCalendar.tsx]: Can this also be abstracted to a type alias?  e.g.
```ts
type DateArray = [number, number, number]
```

File: src/components/Calendar/Toolbar/ExportCalendar.tsx (hunk 2 above)
Comment (ChaseC99) [reply src/components/Calendar/Toolbar/ExportCalendar.tsx]: The type should probably also have a comment explaining that it's `[YEAR, MONTH, DAY, HOUR, MINUTE]`

File: src/termData.ts [hunk 4]
```diff
@@ -67,3 +72,5 @@ function getDefaultTerm() {
 }
 
 export { termData, defaultTerm, getDefaultTerm };
+
+//https://tenor.com/view/leonardo-dicaprio-rick-dalton-point-finger-pointing-hand-right-there-gif-20349799
```
Comment (ChaseC99) [inline src/termData.ts]: what lmao

File: src/components/Calendar/Toolbar/CustomEventDialog/CustomEventDialog.tsx [hunk 5]
```diff
@@ -14,32 +14,58 @@ import {
...
+
+interface CustomEventDialogState {
+    open: boolean,// the properties under this are copies of CustomEvent properties.
+    title: string,
+    start: string,
+    end: string,
+    days: boolean[],
+    scheduleIndices: number[],
+    customEventID: number
+}
```
Comment (ChaseC99) [inline src/components/Calendar/Toolbar/CustomEventDialog/CustomEventDialog.tsx]: Can we just extend CustomEvent?
```suggestion
interface CustomEventDialogState extends CustomEvent {
    open: boolean
}
```

File: src/components/Calendar/Toolbar/CustomEventDialog/CustomEventDialog.tsx (hunk 1 above)
Comment (EricPedley) [reply src/components/Calendar/Toolbar/CustomEventDialog/CustomEventDialog.tsx]: yeah they are.

File: src/components/Calendar/Toolbar/CustomEventDialog/CustomEventDialog.tsx (hunk 5 above)
Comment (EricPedley) [reply src/components/Calendar/Toolbar/CustomEventDialog/CustomEventDialog.tsx]: good idea

File: src/termData.ts (hunk 4 above)
Comment (EricPedley) [reply src/termData.ts]: was pasting that somewhere else and accidentally put it in the project I guess 🥴

File: src/components/Calendar/Toolbar/CustomEventDialog/CustomEventDialog.tsx [hunk 6]
```diff
@@ -14,32 +14,52 @@ import {
...
-    state = {
+interface CustomEventDialogProps {
+    customEvent?: CustomEvent
+    onDialogClose?: ()=>void
+    currentScheduleIndex: number
+    scheduleNames: string[]
+};
+
+interface CustomEventDialogState extends CustomEvent{
+    open: boolean// the properties under this are copies of CustomEvent properties.
```
Comment (ChaseC99) [inline src/components/Calendar/Toolbar/CustomEventDialog/CustomEventDialog.tsx]: This comment can be removed now
```suggestion
    open: boolean
```

**Review by ChaseC99 — APPROVED** [review]
Small nit with the comment but other than that, this looks good to go!

Great work 👏

**Pull request merged**